### PR TITLE
fix(ingress-class):  Add support of custom ingress class

### PR DESCRIPTION
**Pull request by adutchak**
## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## How has this been tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- `helm unittest --helm3 --strict --file 'tests/*.yaml' --file 'tests/clickhouse-operator/*.yaml' charts/posthog`
- `helm lint --strict --set “cloud=local” charts/posthog`

1. Since kubernetes v 1.18 `kubernetes.io/ingress.class` ingress annotation has been [deprecated](https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/#specifying-the-class-of-an-ingress). Instead, a `spec.ingressClassName` was introduced. This PR keeps still `kubernetes.io/ingress.class` annotation for backward compatibility and introduces `spec.ingressClassName`.
2. Considering the above deprecation, it's not possible today to set a custom ingress class, in kubernetes environment with version > 1.18
3. Fixes https://github.com/PostHog/charts-clickhouse/issues/293

## Checklist
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: charts/posthog/ALL_VALUES.md [hunk 1]
```diff
@@ -114,6 +114,7 @@ The following table lists the configurable parameters of the PostHog chart and t
 | ingress.nginx.redirectToTLS | bool | `true` | Whether to redirect to TLS with nginx ingress. |
 | ingress.annotations | object | `{}` | Extra annotations |
 | ingress.secretName | string | `nil` | TLS secret to be used by the ingress. |
+| ingress.class | string | `nil` | A custom ingress class |
```
Comment (hazzadous) [inline charts/posthog/ALL_VALUES.md:117]: Can we say that this "Specifies the ingressClassName used. Defaults to `nginx` if `ingress.type` is `nginx` or `gcp` if `ingress.type` is `clb`"

Feels a bit messy, @guidoiaquinti any thoughts?

**Review by hazzadous — CHANGES_REQUESTED** [review]
Trying to get my head around the different cases!

There's a template error I think, and then some test additions/updates

**Comment by hazzadous** [comment]
Also thanks for the pr 🙏 ping me if you need anything

**Comment by guidoiaquinti** [comment]
👋 Hey @adutchak , are you planning to keep working on this?

**Comment by guidoiaquinti** [comment]
I'm closing this as stale. Feel free to re-open once ready.

**Pull request closed without merging**